articles/ros2_threat_model.md

```diff
@@ -3300,104 +3550,1516 @@ Over time, new reference platforms may be introduced in addition to
+    <td class="danger">✓</td>
+    <td class="success">✘</td>
+    <td class="warning">✘/✓</td>
+    <td>An attacker intercepts the communication to the deployment server and serves malicious content to the devices.
+</td>
+    <td><ul><li>Validate the deployment server through Public Key Infrastructure.</li> <li>Prevent insecure connections to the server from the devices through HTTPS and HSTS policies.</li>
+<li>Digitally sign the binaries sent to the devices.</li></ul>
+</td>
+    <td class="success">Risk is mitigated.</td>
+    <td></td>
+  </tr>
+
+  <tr>
+    <td>An attacker intercepts the binaries sent by the depoyment service.</td>
+    <td class="danger">✓</td>
+    <td class="danger">✓</td>
+    <td class="success">✘</td>
+    <td class="success">✘</td>
+    <td class="success">✘</td>
+    <td class="success">✘</td>
+    <td class="danger">✓</td>
+    <td class="success">✘</td>
+    <td class="warning">✘/✓</td>
+    <td>An attacker intercepts the update communication and stores the binary sent to the devices, gaining access to intellectual property.
+</td>
+    <td><ul><li>Make use of secure, encrypted communication channels.</li> <li>Verify client devices through client certificates.</li>
+<li>Sign and Encrypt update files.</li></ul>
+</td>
+    <td class="success">Risk is mitigated.</td>
+    <td></td>
+  </tr>
+
+  <tr>
+    <td>An attacker prevents the robot  and the deployment service from communicating.</td>
+    <td class="danger">✓</td>
+    <td class="danger">✓</td>
+    <td class="success">✘</td>
+    <td class="success">✘</td>
+    <td class="success">✘</td>
+    <td class="success">✘</td>
+    <td class="danger">✓</td>
+    <td class="success">✘</td>
+    <td class="warning">✘/✓</td>
+    <td>An attacker blocks the robots update process.</td>
+    <td>
+      <ul>
+        <li>Deploy multiple update server endpoints.</li>
+        <li>Deploy a distributed update system.</li>
+      </ul>
+    </td>
+    <td class="warning">Risk is partially mitigated.</td>
+    <td></td>
+  </tr>
+
+</table>
+</div>
+
+#### Attack Trees
+
+In an atempt to analyze the different possible attacks before even happening, attack trees are created. This diagrams speculate about the possible attacks against the system in order to be able to counter them. Threats can be organized in the following [attack trees][wikipedia_attack_tree]. Attacks are ordered starting from the physical attack vector, continuing with network based attacks, and finishing with the support infrastructure for the robot.
+
+#### Physical vector attack tree
+
+The following attack tree describes the possible paths to be followed by an attacker for compromising the system.
+
+![Physical vector attack tree](ros2_threat_model/attack_tree_physical.png)
+
+[Diagram Source (draw.io)](ros2_threat_model/attack_tree_physical.xml)
+
+The next diagram shows the infrastructure affected on a possible attack based on a compromise of a physical communication port.
```

Comment:
affected by?

Reply:
To me it sounds better "the affected infrastructure". Opinions?

Reply:
Sure, SGTM